zellij-server/src/screen.rs

```diff
@@ -1317,6 +1331,60 @@ pub(crate) fn screen_thread_main(
                 screen.render();
                 screen.unblock_input();
             },
+            ScreenInstruction::UpdateSearch(c, client_id) => {
+                if let Some(active_tab) = screen.get_active_tab_mut(client_id) {
+                    active_tab.update_search_term(c, client_id);
+                } else {
+                    log::error!("Active tab not found for client id: {:?}", client_id);
+                }
+
+                screen.render();
+            },
+            ScreenInstruction::SearchForward(client_id) => {
+                if let Some(active_tab) = screen.get_active_tab_mut(client_id) {
+                    active_tab.search_forward(client_id);
+                } else {
+                    log::error!("Active tab not found for client id: {:?}", client_id);
+                }
+
+                screen.render();
+            },
+            ScreenInstruction::SearchBackward(client_id) => {
+                if let Some(active_tab) = screen.get_active_tab_mut(client_id) {
+                    active_tab.search_backward(client_id);
+                } else {
+                    log::error!("Active tab not found for client id: {:?}", client_id);
+                }
+
+                screen.render();
+            },
+            ScreenInstruction::SearchToggleCaseSensitivity(client_id) => {
+                if let Some(active_tab) = screen.get_active_tab_mut(client_id) {
+                    active_tab.toggle_search_case_sensitivity(client_id);
+                } else {
+                    log::error!("Active tab not found for client id: {:?}", client_id);
+                }
+
+                screen.render();
+            },
+            ScreenInstruction::SearchToggleWrap(client_id) => {
+                if let Some(active_tab) = screen.get_active_tab_mut(client_id) {
+                    active_tab.toggle_search_wrap(client_id);
+                } else {
+                    log::error!("Active tab not found for client id: {:?}", client_id);
+                }
+
+                screen.render();
+            },
+            ScreenInstruction::SearchToggleWholeWord(client_id) => {
+                if let Some(active_tab) = screen.get_active_tab_mut(client_id) {
```

Comment:
We've recently added an `active_tab` macro which you can use here to eliminate the boilerplate

Reply:
Ah, cool!